metricbeat/metricbeat.yml

```diff
@@ -21,6 +21,25 @@ metricbeat.config.modules:
 
 #==================== Elasticsearch template setting ==========================
 
+# Example config for templates
+# 
+# When `setup.template.templates` is configured, the old configuration is ignored.
+#setup.template.name: "metricbeat"
+#setup.template.pattern: "metricbeat*"
+#setup.template.fields: "${path.config}/fields.yml"
+#setup.template.enabled: false 
+#setup.template.overwrite: true
+setup.template.templates:
```

Comment:
I would use `setup.templates:`

Reply:
I wanted to have it under the same namespace. However, this configs were mainly meant to allow playing around with multiple templates for now and show how they can be used.
As there are ongoing discussions around how to group together configs that belong together, https://github.com/elastic/beats/pull/7963#discussion_r237654120, we might end up putting the `templates` under some completely different group. 
My focus was on the internal implementation and we could still keep the logic I built when the user facing config changes. That's actually the reason I introduced the `Setup` https://github.com/elastic/beats/pull/9247/files/7ff9c687751ca970678a2856fe19fc14ec6f1356#diff-b01453e5fd277f7039381e366330a2d2R24 method.

Reply:
Concerning the naming @urso and myself had few followup discussion where it should be, but not clear plan yet. 

I think `setup.templates` is good choice.

Reply:
We probably need to change it again when combining with ILM. Will think about it a bit more.